src/back/spirv/layout.rs

```diff
@@ -0,0 +1,216 @@
+use spirv::*;
+
+pub(crate) struct Module {
```

Comment:
I guess your suggestions means that all content of this file has to be moved to `mod.rs` and removing the pub(crate), so it will still work?

I tried it, and that works, but wouldn't that remove the benefit of having multiple files to split the code? As I am used to OOP programming, there it is usually standard practice to split all clases to their own files. Is this different in Rust?

Reply:
Similar to how the IR is set up: you don't need to move the logic into `mod.rs`, just the definitions that are used across different sub-modules. The logic itself can be in sub-modules, e.g. `impl super::MyStruct { ... }`. You still have multiple sub-modules with this approach.
The benefit of this is that these types are not visible to other front-ends and backends.

Reply:
Done that, looking good!